akka-stream/src/main/scala/akka/stream/impl/io/TcpListenStreamActor.scala

```diff
@@ -99,7 +99,6 @@ private[akka] class TcpListenStreamActor(localAddressPromise: Promise[InetSocket
         pump()
       case f: CommandFailed ⇒
         val ex = new ConnectionException(s"Command [${f.cmd}] failed")
-        unbindPromise.tryFailure(ex)
```

Comment:
I need to check the reason for the CommandFailed. If it was a result of an Unbind then I actually need to fail here.